### PR TITLE
Correct the return value on match and mismatch for MAC pkeys

### LINKED ISSUE #14147: EVP_PKEY_cmp() Fails Identical keys
Create a EVP_PKEY and compare it to itself. EVP_PKEY_cmp() returns 0, which means a mismatch.
See code fragment below.

```
	int rv = 0;

	EVP_PKEY *mac_key = EVP_PKEY_new_raw_private_key(EVP_PKEY_HMAC, NULL,
	                                                 (const unsigned char *)("0123456789"),
	                                                 strlen("0123456789"));

    rv = EVP_PKEY_cmp(mac_key, mac_key);
    if (rv != 1) {
        printf("key compare failed. rv = %d\n", rv);
        return 0;
    }
    printf("key compare passed\n");
```

Possible fix ...
```
@ -130,8 +130,14 @@ int EVP_PKEY_cmp(const EVP_PKEY *a, const EVP_PKEY *b)
                return ret;
        }

        if (a->ameth->pub_cmp)
            return a->ameth->pub_cmp(a, b);
        if (a->ameth->pub_cmp) {
            ret = a->ameth->pub_cmp(a, b);
            if (ret == 0) {
                return 1;
            } else {
                return 0;
            }
        }
    }
```
The issue is present in the latest 1.1.1 stable branch. And appears to be on Master (3.0) as well.

### PR DESCRIPTION
**Pull request by t8m**
The pub_cmp function must return 1 on match and 0 otherwise.

Fixes #14147

##### Checklist

- [x] tests are added or updated

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

File: test/evp_extra_test.c [hunk 1]
```diff
@@ -873,6 +881,7 @@ static int test_set_get_raw_keys_int(int tst, int pub)
     }
 
     if (!TEST_ptr(pkey)
+            || !TEST_int_eq(EVP_PKEY_cmp(pkey, pkey), 1)
```
Comment (slontis) [inline test/evp_extra_test.c:884]: This should use EVP_PKEY_eq()..

File: crypto/hmac/hm_ameth.c [hunk 2]
```diff
@@ -47,7 +47,8 @@ static int hmac_pkey_ctrl(EVP_PKEY *pkey, int op, long arg1, void *arg2)
 
 static int hmac_pkey_public_cmp(const EVP_PKEY *a, const EVP_PKEY *b)
```
Comment (slontis) [inline crypto/hmac/hm_ameth.c:48]: use hmac_pkey_public_eq ??

File: test/evp_extra_test.c (hunk 1 above)
Comment (t8m) [reply test/evp_extra_test.c:884]: This is on 1.1.1 branch, no EVP_PKEY_eq there.

File: crypto/hmac/hm_ameth.c (hunk 2 above)
Comment (t8m) [reply crypto/hmac/hm_ameth.c:48]: This is 1.1.1 branch. I do not think we want such kind of changes there.

**Comment by t8m** [comment]
Merged to master. Thank you for the review.

**Pull request closed without merging**